src/microstate.js

```diff
@@ -24,30 +26,56 @@ export default class MicroState {
 
         let descriptor = getOwnPropertyDescriptor(attrs, name);
         if (descriptor.get && typeof descriptor.get === 'function') {
+          // property is a getter
           defineProperty(prototype, name, descriptor);
+        } else if (name === 'transitions' && typeof descriptor.value === 'object') {
+          // property is transitions hash
+          let transitions = assign({}, descriptor.value, {
+            replace(current, ...args) {
+              return new this.constructor(...args);
+            }
+          });
+          getOwnPropertyNames(transitions).forEach(function(name){
+            let descriptor = getOwnPropertyDescriptor(transitions, name);
+            defineProperty(prototype, name, makeTransitionDescriptor(descriptor));
+          });
         } else if (name === 'valueOf' || name === 'toString') {
```

Comment:
Once we remove the the `prototype` property, now that we have the `transitions` hash, I think we can get rid of this arm of the `if`. I think this is the only thing keeping this from getting merged.